gbpservice/neutron/nsf/db/nsf_db.py

```diff
@@ -0,0 +1,405 @@
+                session,
+                nsf_db_model.NetworkServiceInstance,
+                network_service_instance_id)
+        except exc.NoResultFound:
+            raise nsf_exc.NetworkServiceInstanceNotFound()
+
+    def update_network_service_instance(self, session,
+                                        network_service_instance_id,
+                                        updated_network_service_instance):
+        with session.begin(subtransactions=True):
+            network_service_instance_db = self._get_network_service_instance(
+                session, network_service_instance_id)
+            if 'port_info' in updated_network_service_instance:
+                self._set_port_info_for_nsi(
+                    session,
+                    network_service_instance_db,
+                    updated_network_service_instance)
+            network_service_instance_db.update(
+                updated_network_service_instance)
+        return self._make_network_service_instance_dict(
+            network_service_instance_db)
+
+    def delete_network_service_instance(self, session,
+                                        network_service_instance_id):
+        with session.begin(subtransactions=True):
+            network_service_instance_db = self._get_network_service_instance(
+                session, network_service_instance_id)
+            session.delete(network_service_instance_db)
+
+    def get_network_service_instance(self, session,
+                                     network_service_instance_id,
+                                     fields=None):
+        network_service_instance = self._get_network_service_instance(
+            session, network_service_instance_id)
+        return self._make_network_service_instance_dict(
+            network_service_instance, fields)
+
+    def get_network_service_instances(self, session, filters=None, fields=None,
+                                      sorts=None, limit=None, marker=None,
+                                      page_reverse=False):
+        marker_obj = self._get_marker_obj(
+            'network_service_instances', limit, marker)
+        return self._get_collection(
+            session, nsf_db_model.NetworkServiceInstance,
+            self._make_network_service_instance_dict,
+            filters=filters, fields=fields, sorts=sorts, limit=limit,
+            marker_obj=marker_obj, page_reverse=page_reverse)
+
+    def _set_mgmt_ports_for_nsd(self, session, network_service_device_db,
+                                network_service_device):
+        nsd_db = network_service_device_db
+        mgmt_data_ports = network_service_device.get('mgmt_data_ports')
+        if not mgmt_data_ports:
+            nsd_db.mgmt_data_ports = []
+            return
+        with session.begin(subtransactions=True):
+            nsd_db.mgmt_data_ports = []
+            for port in mgmt_data_ports:
+                port_info_db = nsf_db_model.PortInfo(
```

Comment:
It was kept as a placeholder for HA. Will remove it, and can add back as per HA design